extension/trace.h

```diff
@@ -161,7 +159,13 @@ static zend_always_inline int begin_profiling(zend_string *root_symbol, zend_exe
     } else {
 #if PHP_VERSION_ID >= 80000
         hp_entry_t *cur_entry = hp_fast_alloc_hprof_entry();
-        (cur_entry)->name_hprof = zend_string_copy((*(entries))->name_hprof);
+        /* Check if entries is not NULL before dereferencing */
```

Comment:
Previously assumed `*(entries)` is always non-NULL when creating entries for ignored functions.